tests/utils/scripts/stats_parsers.py

```diff
@@ -0,0 +1,105 @@
+import statistics
+
+
+class StatisticsParser:
+    executor = None
+
+    def raise_not_implemented(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def _set_executor(self, stats: list):
+        if self.executor:
+            return
+        else:
+            for executor in [TimeStatisticsParser, MemoryStatisticsParser]:
+                try:
+                    executor().parse_stats(stats)
+                    self.executor = executor()
+                    break
+                except:
+                    self.executor = None
+            if not self.executor:
+                self.raise_not_implemented()
+
+    def parse_stats(self, stats: list):
+        self._set_executor(stats)
+        getattr(self.executor, "parse_stats", self.raise_not_implemented)(stats)
+
+    def append_stats(self, stats: list):
+        self._set_executor(stats)
+        getattr(self.executor, "append_stats", self.raise_not_implemented)(stats)
+
+    def aggregate_stats(self):
+        getattr(self.executor, "aggregate_stats", self.raise_not_implemented)()
+
+
+class TimeStatisticsParser:
+
+    def __init__(self):
+        self.last_stats = {}
+        self.combined_stats = {}
+        self.aggregated_stats = {}
+
+    def parse_stats(self, stats: list):
+        """Parse raw statistics from nested list to flatten dict"""
+        for element in stats:
+            if isinstance(element, (int, float)):
+                for k, v in self.last_stats.items():
+                    if v is None:
+                        self.last_stats.update({k: [element]})
+            else:
+                for k, v in element.items():
+                    if len(v) == 1:
+                        self.last_stats.update({k: v})
+                    else:
+                        self.last_stats.update({k: None})
+                        self.parse_stats(v)
+
+    def append_stats(self, stats: list):
+        self.parse_stats(stats)
+        if not self.combined_stats:
+            self.combined_stats = self.last_stats.copy()
+            return
+        for step_name, duration in self.last_stats.items():
+            self.combined_stats[step_name].extend(duration)
+
+    def aggregate_stats(self):
+        """Aggregate provided statistics"""
+        self.aggregated_stats = {step_name: {"avg": statistics.mean(duration_list),
+                                             "stdev": statistics.stdev(duration_list) if len(duration_list) > 1 else 0}
+                                 for step_name, duration_list in self.last_stats.items()}
+
+
+class MemoryStatisticsParser:
+
+    def __init__(self):
+        self.last_stats = {}
+        self.combined_stats = {}
+        self.aggregated_stats = {}
+
+    def parse_stats(self, stats: list, key: str = ""):
```

Comment:
update `parse_stats`, `append_stats` and `aggregate_stats` according to removing of nested values